app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Users
+  class SessionsController < Devise::SessionsController
+    def new
+      if request.referer.present?
+        referer = URI.parse(request.referer)
+        if referer.host == request.host
+          self.resource = resource_class.create(sign_in_params)
+          store_location_for(resource, request.referer)  
+        end
+      end
+    rescue URI::InvalidURIError
+      # bad URI, ignore
+    ensure
+      super
+    end
```

Comment:
I updated this code to always use `request.referer` to access the referer and to check that the hosts are equal (before it was comparing the entire referer to just the host.